tests/raft_log_replication.py

```diff
@@ -593,7 +593,419 @@ async def _unittest_raft_log_replication() -> None:
+    assert raft_node_1._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_1._log[0].entry.value == 0
+    assert raft_node_1._log[1].term == 4
+    assert raft_node_1._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_1._log[1].entry.value == 7
+    assert raft_node_1._log[2].term == 5
+    assert raft_node_1._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_1._log[2].entry.value == 8
+    assert raft_node_1._log[3].term == 6
+    assert raft_node_1._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_1._log[3].entry.value == 9
+    assert raft_node_1._commit_index == 3
+
+    # check if the new entry is replicated in the follower nodes
+    assert len(raft_node_2._log) == 1 + 3
+    assert raft_node_2._log[0].term == 0
+    assert raft_node_2._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_2._log[0].entry.value == 0
+    assert raft_node_2._log[1].term == 4
+    assert raft_node_2._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_2._log[1].entry.value == 7
+    assert raft_node_2._log[2].term == 5
+    assert raft_node_2._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_2._log[2].entry.value == 8
+    assert raft_node_2._log[3].term == 6
+    assert raft_node_2._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_2._log[3].entry.value == 9
+    assert raft_node_2._commit_index == 3
+
+    assert len(raft_node_3._log) == 1 + 3
+    assert raft_node_3._log[0].term == 0
+    assert raft_node_3._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_3._log[0].entry.value == 0
+    assert raft_node_3._log[1].term == 4
+    assert raft_node_3._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_3._log[1].entry.value == 7
+    assert raft_node_3._log[2].term == 5
+    assert raft_node_3._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_3._log[2].entry.value == 8
+    assert raft_node_3._log[3].term == 6
+    assert raft_node_3._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_3._log[3].entry.value == 9
+    assert raft_node_3._commit_index == 3
+    
+    await asyncio.sleep(TERM_TIMEOUT + 1)
+    _logger.info("================== TEST 1: Change LEADER and check logs ==================")
+
+    # New LEADER => raft_node_2
+
+    raft_node_1.election_timeout = ELECTION_TIMEOUT + 1
+    raft_node_2.election_timeout = ELECTION_TIMEOUT
+    raft_node_3.election_timeout = ELECTION_TIMEOUT + 1
+
+    raft_node_1._change_state(RaftState.FOLLOWER)
+    raft_node_2._change_state(RaftState.FOLLOWER)
+    raft_node_3._change_state(RaftState.FOLLOWER)
+    raft_node_1._voted_for = None
```

Comment:
Why is this necessary? (Should't it be able to know that it hasn't voted for anyone just be being made a Follower?)